tests/unit/fs_mock_metadata_to_disk_buffer_test.cc

```diff
@@ -0,0 +1,398 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+#include "fs/metadata_disk_entries.hh"
+#include "fs/metadata_log.hh"
+#include "fs_mock_metadata_to_disk_buffer.hh"
+#include "mock_block_device.hh"
+
+#include <seastar/core/print.hh>
+#include <seastar/core/temporary_buffer.hh>
+#include <seastar/core/units.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/test_runner.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include <stdint.h>
+
+using namespace seastar;
+using namespace seastar::fs;
+
+namespace {
+
+using append = mock_metadata_to_disk_buffer::action::append;
+using flush_to_disk = mock_metadata_to_disk_buffer::action::flush_to_disk;
+constexpr mock_metadata_to_disk_buffer::append_result APPENDED = mock_metadata_to_disk_buffer::APPENDED;
+constexpr mock_metadata_to_disk_buffer::append_result TOO_BIG = mock_metadata_to_disk_buffer::TOO_BIG;
+
+constexpr size_t default_buff_size = 1 * MB;
+constexpr size_t default_alignment = 4 * KB;
+constexpr size_t checkpoint_size = sizeof(ondisk_type) + sizeof(ondisk_checkpoint);
+
+}
+
+// the following test checks if exceeding append is properly handled
+SEASTAR_THREAD_TEST_CASE(size_exceeded_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+    constexpr size_t delete_inode_size = sizeof(ondisk_type) + sizeof(ondisk_delete_inode);
+    constexpr size_t next_metadata_cluster_size = sizeof(ondisk_type) + sizeof(ondisk_next_metadata_cluster);
+
+    size_t remaining_space = default_buff_size - checkpoint_size - next_metadata_cluster_size;
+    while (true) {
```

Comment:
Maybe you would want to use this:
```suggestion
    for (;;) {
```
for infinite loops?